### PR TITLE
change property `p` for event from SVGPoint to SVG.Point - fixing #57

### PR DESCRIPTION
**Pull request by dotnetCarpenter**
Instead of doing:

```js
const point = {x:0,y:0}
rect.on('drawstart', event => {
  const {x,y} = event.detail.p
  point.x = x
  point.y = y
})
```

You can now do:

```js
let point
rect.on('drawstart', event => {
  point = event.detail.p.clone()
})
```

Please see #57 for more info.

### CODE REVIEW

File: src/svg.draw.js [hunk 1]
```diff
@@ -3,12 +3,12 @@
...
         this.el = el;
         el.remember('_paintHandler', this);
-    
+
         var _this = this,
             plugin = this.getPlugin();
 
         this.parent = el.parent(SVG.Nested) || el.parent(SVG.Doc);
-        this.p = this.parent.node.createSVGPoint(); // Helping point for coord transformation
+        this.p = new SVG.Point(this.parent.node.createSVGPoint()); // Helping point for coord transformation
```
Comment (dotnetCarpenter) [inline src/svg.draw.js]: I'm not sure if we need to use `node.createSVGPoint()` as constructor argument. In my tests it works fine to use nothing - both results in x:0, y:0. But I did not test with the parent node being transformed. Does it make a difference for `matrixTransform`?

Comment (Fuzzyma) [reply src/svg.draw.js]: Nope you dont need to do that. The reason a native point was used here is that there simply wasnt no SVG.Point when I wrote this plugin :).

**Comment by dotnetCarpenter** [comment]
>The reason a native point was used here is that there simply wasnt no SVG.Point when I wrote this plugin :)

@Fuzzyma Should `event.detail.m` be an instance of [SVG.Matrix](https://svgjs.com/docs/2.7/classes/#svg-matrix) then?

There is a small overhead of using the class instead of a native [`DOMMatrix`](https://developer.mozilla.org/en-US/docs/Web/API/DOMMatrix) (`getScreenCTM()`). But it makes sense that both properties are svg.js classes. I'm just not sure what improvement it will give. I don't have a use-case.

File: src/lineable.js [hunk 2]
```diff
@@ -85,7 +85,7 @@
                 this.p.x = array[i][0]
                 this.p.y = array[i][1]
 
-                var p = this.p.matrixTransform(this.parent.node.getScreenCTM().inverse().multiply(this.el.node.getScreenCTM()));
+                var p = this.p.native().matrixTransform(this.parent.node.getScreenCTM().inverse().multiply(this.el.node.getScreenCTM()));
```
Comment (Fuzzyma) [inline src/lineable.js:88]: just use `transform()` of SVG.Point here and pass it `this.parent.screenCTM()...` and multiply with `this.el.screenCTM()`.

File: src/svg.draw.js [hunk 3]
```diff
@@ -47,15 +47,15 @@
 
         this.p.x = x - (this.offset.x - window.pageXOffset);
         this.p.y = y - (this.offset.y - window.pageYOffset);
-        
-        return this.p.matrixTransform(this.m);
-    
+
+        return this.p.native().matrixTransform(this.m);
```
Comment (Fuzzyma) [inline src/svg.draw.js:51]: `this.p.transform(...)`

File: src/svg.draw.js [hunk 4]
```diff
@@ -47,15 +47,15 @@
...
-    
+
     PaintHandler.prototype.start = function (event) {
-    
+
         var _this = this;
-    
+
         // get the current transform matrix from screen to element (offset corrected)
         this.m = this.el.node.getScreenCTM().inverse();
```
Comment (Fuzzyma) [inline src/svg.draw.js:60]: `this.el.screenCTM()...`

**Review by Fuzzyma — CHANGES_REQUESTED** [review]
Lets just move to svg.js objects only. I think this functionality wsnt available when I wrote teh plugin but there is no poin in keeping old code around.

**Comment by Fuzzyma** [comment]
Alright, I will put this on the backlog ^^

**Comment by dotnetCarpenter** [comment]
I might have an hour now to fix this

**Comment by dotnetCarpenter** [comment]
Sorry, no I do not have time now. But I will try to fix this, this week.

Just to be clear. This can be merged when `SVG.Matrix` is being used instead of `this.el.node.getScreenCTM().inverse()` and use `SVG.Point.transform(this.m)` instead of `this.p.native().matrixTransform(this.m)`.

And lastly, we can use `this.m = this.el.getScreenCTM().inverse()` instead of `this.m = this.el.node.getScreenCTM().inverse()`.

I might throw in an example folder to manually test the new code is working...

**Comment by Fuzzyma** [comment]
One typo: in svg.js its screenCTM and not getScreenCTM. But no pressure. This plugin needs a rework at some point anyway. Github just launched its new notification dashboard and this PR came up. So I want this merged, release it and then make the whole plugin compatible to svg.js v3

**Pull request closed without merging**